docs/proposals/karmada-operator/custom_ca_cert/README.md

```diff
@@ -0,0 +1,122 @@
+
+## Motivation
+
+In high-availability scenarios where Karmada control planes power mission-critical workloads, ensuring that each managed control plane adheres to strict organizational policies around availability
+and disaster recovery is essential. Deploying control planes across multiple management clusters that span multiple data centers provides redundancy and resilience in the event of a data center outage. 
+However, when multiple control plane instances must access a shared etcd instance and present a unified API endpoint, using a common CA certificate across instances is necessary. 
+By enabling users to specify a custom CA certificate, this feature ensures that control planes spanning data centers function as a cohesive unit while meeting security and availability standards.
+
+### Architecture Overview
+![architecture-overview](./architecture-overview.png)
+
+### Goals
+
+- Allow users to specify a custom CA certificate for the Karmada control plane.
+- Ensure that control plane instances deployed across multiple management clusters that span multiple data centers use the same CA, enabling secure and seamless cross-data center communication.
+- Enable operators to align Karmada control plane PKI with organizational policies for high availability and security.
+
+### Non-Goals
+
+- Change the default behavior of the Karmada operator when no custom CA is provided.
+- Alter the process of how control planes are connected to etcd.
+- Address custom CA requirements for other Karmada components.
+
+## Proposal
+
+The proposal introduces a new optional field, `CertificateConfig`, in the `KarmadaSpec`, where users can specify a custom CA certificate for the Karmada control plane. 
+This configuration will allow users to choose between `Auto` and `Custom` modes:
+
+1. *Auto Mode*: The operator will automatically generate a CA certificate for the control plane. This mode is the default.
+2. *Custom Mode*: The user can specify a Kubernetes secret containing their CA certificate and key to be used for the control plane.
+
+### API Changes
+
+```go
+// CertificateMode defines the mode for managing certificates for the Karmada control plane.
+type CertificateMode string
+
+const (
+   // Auto mode means the certificate will be automatically generated by the operator.
+   Auto CertificateMode = "Auto"
+   
+   // Custom mode means the user provides their own CA certificate.
+   Custom CertificateMode = "Custom"
+)
+
+// CustomCertConfig defines the structure for user-provided CA certificates in Custom mode.
+type CustomCertConfig struct {
+   // CACert references a Kubernetes secret containing the CA certificate for the Karmada instance.
+   // The secret must contain the following data keys:
+   // tls.crt: The TLS certificate.
+   // tls.key: The TLS private key.
+   CACert *LocalSecretReference `json:"caCert,omitempty"`
+}
```

Comment:
Is there a possibility to add other fields to customize other certificates?
I don't know, just asking and thinking if there is any interaction with the certification rotation feature in the future.

Reply:
Yes. Absolutely. Right now, this is very minimal, but can be extended in the future.